### ISSUE TITLE
win10,执行train_multi_machine.py出现Operator "gen_nccl_id" has not been registered

### ISSUE THREAD

**Issue by zhouyuxixixi** [PaddlePaddle/PaddleDetection#1917]
win10+paddlepaddle2.0.0rc+paddledetection-master

执行命令为
python -m paddle.distributed.launch --selected_gpus="0"  tools/train_multi_machine.py -c configs/ppyolo/ppyolo_r18vd_my.yml --eval
报错
Operator "gen_nccl_id" has not been registered
![image](https://user-images.githubusercontent.com/65207014/102461320-2e808480-4083-11eb-94c5-cbd4f6206f98.png)
**yml 文件如下**
architecture: YOLOv3
use_gpu: true
max_iters: 2000
log_iter: 10
save_dir: output
snapshot_iter: 100
metric: VOC
pretrain_weights:
weights: output/ppyolo_tiny/model_final
num_classes: 1
use_fine_grained_loss: true
use_ema: true
ema_decay: 0.9998

YOLOv3:
  backbone: ResNet
  yolo_head: YOLOv3Head
  use_fine_grained_loss: true

ResNet:
  norm_type: sync_bn
  freeze_at: 0
  freeze_norm: false
  norm_decay: 0.
  depth: 18
  feature_maps: [4, 5]
  variant: d

YOLOv3Head:
  anchor_masks: [[3, 4, 5], [0, 1, 2]]
  anchors: [[19, 24], [18, 25], [20, 23],
            [18, 28], [20, 27], [21, 26]]
  norm_decay: 0.
  conv_block_num: 0
  scale_x_y: 1.05
  yolo_loss: YOLOv3Loss
  nms: MatrixNMS
  drop_block: true

YOLOv3Loss:
  ignore_thresh: 0.7
  scale_x_y: 1.05
  label_smooth: false
  use_fine_grained_loss: true
  iou_loss: IouLoss

IouLoss:
  loss_weight: 2.5
  max_height: 608
  max_width: 608

MatrixNMS:
    background_label: -1
    keep_top_k: 100
    normalized: false
    score_threshold: 0.01
    post_threshold: 0.01

LearningRate:
  base_lr: 0.004
  schedulers:
  - !PiecewiseDecay
    gamma: 0.1
    milestones:
    - 150000
    - 200000
  - !LinearWarmup
    start_factor: 0.
    steps: 4000

OptimizerBuilder:
  optimizer:
    momentum: 0.9
    type: Momentum
  regularizer:
    factor: 0.0005
    type: L2

_READER_: 'ppyolo_reader_my.yml'
TrainReader:
  inputs_def:
    fields: ['image', 'gt_bbox', 'gt_class', 'gt_score']
    num_max_boxes: 20
  dataset:
    !VOCDataSet
      anno_path: train_detection.txt
      dataset_dir: E:\\DeepLearning\\DataSet\\Cigarette\\OnlyCigarettePaddleDetection
      use_default_label: False
      with_background: false
  sample_transforms:
    - !DecodeImage
      to_rgb: True
    - !ResizeImage
      target_size: 224
      max_size: 224
    - !NormalizeBox {}
    - !PadBox
      num_max_boxes: 20
    - !BboxXYXY2XYWH {}
  batch_transforms:
  - !RandomShape
    sizes: [224]
    random_inter: True
  - !NormalizeImage
    mean: [0.485, 0.456, 0.406]
    std: [0.229, 0.224, 0.225]
    is_scale: True
    is_channel_first: false
  - !Permute
    to_bgr: false
    channel_first: True
  - !Gt2YoloTarget
    anchor_masks: [[3, 4, 5], [0, 1, 2]]
    anchors: [[19, 24], [18, 25], [20, 23],
              [18, 28], [20, 27], [21, 26]]
    downsample_ratios: [32, 16]
  batch_size: 4
  shuffle: true
  mixup_epoch: 500
  drop_last: true
  worker_num: 0
  bufsize: 1
  use_process: False

**ppyolo-reader**
TrainReader:
  inputs_def:
    fields: ['image', 'gt_bbox', 'gt_class', 'gt_score']
    num_max_boxes: 50
  dataset:
    !COCODataSet
      image_dir: train2017
      anno_path: annotations/instances_train2017.json
      dataset_dir: dataset/coco
      with_background: false
  sample_transforms:
    - !DecodeImage
      to_rgb: True
      with_mixup: True
    - !MixupImage
      alpha: 1.5
      beta: 1.5
    - !ColorDistort {}
    - !RandomExpand
      ratio: 2.0
      fill_value: [123.675, 116.28, 103.53]
    - !RandomCrop {}
    - !RandomFlipImage
      is_normalized: false
    - !NormalizeBox {}
    - !PadBox
      num_max_boxes: 50
    - !BboxXYXY2XYWH {}
  batch_transforms:
  - !RandomShape
    sizes: [320, 352, 384, 416, 448, 480, 512, 544, 576, 608]
    random_inter: True
  - !NormalizeImage
    mean: [0.485, 0.456, 0.406]
    std: [0.229, 0.224, 0.225]
    is_scale: True
    is_channel_first: false
  - !Permute
    to_bgr: false
    channel_first: True
  - !Gt2YoloTarget
    anchor_masks: [[6, 7, 8], [3, 4, 5], [0, 1, 2]]
    anchors: [[10, 13], [16, 30], [33, 23],
              [30, 61], [62, 45], [59, 119],
              [116, 90], [156, 198], [373, 326]]
    downsample_ratios: [32, 16, 8]
  batch_size: 24
  shuffle: true
  mixup_epoch: 25000
  drop_last: true
  worker_num: 8
  bufsize: 4
  use_process: true

EvalReader:
  inputs_def:
    fields: ['image', 'im_size', 'im_id', 'gt_bbox', 'gt_class', 'is_difficult']
    num_max_boxes: 20
  dataset:
    !VOCDataSet
      anno_path: eval_detection.txt
      dataset_dir: E:\\DeepLearning\\DataSet\\Cigarette\\OnlyCigarettePaddleDetection
      use_default_label: False
      with_background: false
  sample_transforms:
    - !DecodeImage
      to_rgb: True
    - !ResizeImage
      target_size: 224
      max_size: 224
    - !NormalizeImage
      mean: [0.485, 0.456, 0.406]
      std: [0.229, 0.224, 0.225]
      is_scale: True
      is_channel_first: false
    - !PadBox
      num_max_boxes: 20
    - !Permute
      to_bgr: false
      channel_first: True
  batch_size: 1
  drop_empty: false
  worker_num: 0
  bufsize: 1

TestReader:
  inputs_def:
    image_shape: [3, 224, 224]
    fields: ['image', 'im_size', 'im_id']
  dataset:
    !ImageFolder
      anno_path: annotations/instances_val2017.json
      with_background: false
  sample_transforms:
    - !DecodeImage
      to_rgb: True
    - !ResizeImage
      target_size: 608
      interp: 2
    - !NormalizeImage
      mean: [0.485, 0.456, 0.406]
      std: [0.229, 0.224, 0.225]
      is_scale: True
      is_channel_first: false
    - !Permute
      to_bgr: false
      channel_first: True
  batch_size: 1

**Comment by zhouyuxixixi** [comment]
python -m paddle.distributed.launch --selected_gpus="0" tools/train_multi_machine.py -c configs/ppyolo/ppyolo_r18vd_my.yml --eval
**不要使用train_multi_machine.py**
请有尊严的使用**train.py**
